cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,378 @@
+			{
+				groupId:    "com.azure.spring",
+				artifactId: "spring-cloud-azure-starter-data-cosmos",
+			},
+		},
+	},
+}
+
+func updateProjectByAnalyzingSpringBootProject(
+	parentProject *mavenProject, mavenProject *mavenProject, azdProject *Project) {
+	if !isSpringBootApplication(mavenProject) {
+		log.Printf("Skip analyzing spring boot project. path = %s.", mavenProject.path)
+		return
+	}
+	var springBootProject = SpringBootProject{
+		springBootVersion:     detectSpringBootVersion(parentProject, mavenProject),
+		applicationProperties: readProperties(azdProject.Path),
+		parentProject:         parentProject,
+		mavenProject:          mavenProject,
+	}
+	addDatabases(azdProject, &springBootProject)
+	addServiceBus(azdProject, &springBootProject)
+	addEventHubs(azdProject, &springBootProject)
+	addStorageAccount(azdProject, &springBootProject)
+	addSpringCloudAzure(azdProject, &springBootProject)
+}
+
+func addDatabases(azdProject *Project, springBootProject *SpringBootProject) {
+	databaseDepMap := map[DatabaseDep]struct{}{}
+	for _, rule := range databaseDependencyRules {
+		for _, targetDependency := range rule.mavenDependencies {
+			for _, projectDependency := range springBootProject.mavenProject.Dependencies {
+				if projectDependency.GroupId == targetDependency.groupId &&
+					projectDependency.ArtifactId == targetDependency.artifactId {
+					databaseDepMap[rule.databaseDep] = struct{}{}
+					logServiceAddedAccordingToMavenDependency(rule.databaseDep.Display(),
+						projectDependency.GroupId, projectDependency.ArtifactId)
+					break
+				}
+			}
+		}
+	}
+	if len(databaseDepMap) > 0 {
+		azdProject.DatabaseDeps = slices.SortedFunc(maps.Keys(databaseDepMap),
+			func(a, b DatabaseDep) int {
+				return strings.Compare(string(a), string(b))
+			})
+	}
+}
+
+func logServiceAddedAccordingToMavenDependency(resourceName, groupId string, artifactId string) {
+	logServiceAddedAccordingToMavenDependencyAndExtraCondition(resourceName, groupId, artifactId, "")
+}
+
+func logServiceAddedAccordingToMavenDependencyAndExtraCondition(
+	resourceName, groupId string, artifactId string, extraCondition string) {
+	insertedString := ""
+	extraCondition = strings.TrimSpace(extraCondition)
+	if extraCondition != "" {
+		insertedString = " and " + extraCondition
+	}
+	log.Printf("Added '%s' because found dependency '%s:%s'%s.",
+		resourceName, groupId, artifactId, insertedString)
+}
+
+func addServiceBus(azdProject *Project, springBootProject *SpringBootProject) {
+	// we need to figure out multiple projects are using the same service bus
+	addServiceBusAccordingToJMSMavenDependency(azdProject, springBootProject)
+	addServiceBusAccordingToSpringCloudStreamBinderMavenDependency(azdProject, springBootProject)
+}
+
+func addServiceBusAccordingToJMSMavenDependency(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter-servicebus-jms"
```

Comment:
This dependency used for more than once, use variable to make sure they are the same in 2 places.